### ISSUE TITLE
(🐞) Can't parse single quoted doc string longer than line length

### ISSUE THREAD

**Issue by KotlinIsland** [psf/black#3141]
```py
def f():
    "lorem ipsum ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd         sus"
```
```
> black --version
black, 22.6.0 (compiled: yes)
Python (CPython) 3.10.5                   
> black test.py
error: cannot format ...
```

I have `preview` enabled.

**Comment by ichard26** [comment]
OK, so Black is producing invalid code and it's also being misreported as a parse failure when really the root issue Black moved the ending quote to its own line...
```python
def f():
    "lorem ipsum ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd         sus
    "
```
I really hope we can get rid of the two-pass hack soon or at least make the error handling a bit better.

**Comment by ichard26** [comment]
Technically we could resolve this by enforcing triple quotes for docstrings so the buggy logic introduced in https://github.com/psf/black/commit/20d8ccb54253f8a66321f6708d53e2a05a54079b doesn't have worry about single quotes breaking things. Funnily enough there's a comment mentioning this possibility:

https://github.com/psf/black/blob/05b63c4bccbfb292a92d3ec962ce9b8fa4ebcfd5/src/black/linegen.py#L330-L331

... but I'll defer the decision to whether we want to enforce triple quotes for docstrings as docstring handling a touchy area so I'd rather not make decision too quickly. See https://github.com/psf/black/issues/982 for this discussion. 

---

For the time being, I'll fix the code introduced in 20d8ccb54253f8a66321f6708d53e2a05a54079b and maybe improve Black's error handling around here too.
